File: app/src/main/java/net/mmho/photomap2/LicenseAdapter.java
package net.mmho.photomap2;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;

public class LicenseAdapter extends ArrayAdapter<String> {

    private int resource;
    private LayoutInflater inflater;

    public LicenseAdapter(Context context, int resource,String[] objects) {
        super(context, resource,objects);
        inflater = (LayoutInflater) getContext().getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        this.resource = resource;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        View v = convertView;
        if(v==null) v = inflater.inflate(resource, null);
        ((LicenseLayout)v).setData(getItem(position));
        return v;
    }
}

File: app/src/main/kotlin/net/mmho/photomap2/LicenseAdapter.kt
package net.mmho.photomap2

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter

internal class LicenseAdapter(context: Context, private val resource: Int, objects: Array<String>)
        : ArrayAdapter<String>(context, resource, objects) {
    private val inflater: LayoutInflater
        = getContext().getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val v: View = convertView ?: inflater.inflate(resource, null)
        (v as LicenseLayout).setData(getItem(position))
        return v
    }
}
